modules/oe_editorial_unpublish/src/Routing/RouteSubscriber.php

```diff
@@ -16,50 +17,63 @@
 class RouteSubscriber extends RouteSubscriberBase {
 
   /**
-   * The unpublishable content entities service.
+   * The entity type manager.
    *
-   * @var \Drupal\oe_editorial_unpublish\UnpublishableContentEntities
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
-  protected $unpublishableContentEntities;
+  protected $entityTypeManager;
 
   /**
-   * RouteSubscriber constructor.
+   * Constructs an instance of RouteSubscriber.
    *
-   * @param \Drupal\oe_editorial_unpublish\UnpublishableEntitiesInterface $unpublishableContentEntities
-   *   The unpublishable content entities service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
    */
-  public function __construct(UnpublishableEntitiesInterface $unpublishableContentEntities) {
-    $this->unpublishableContentEntities = $unpublishableContentEntities;
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
   }
 
   /**
    * {@inheritdoc}
    */
   protected function alterRoutes(RouteCollection $collection) {
-    $definitions = $this->unpublishableContentEntities->getDefinitions();
+    $definitions = $this->entityTypeManager->getDefinitions();
     if (!$definitions) {
       return;
     }
 
     foreach ($definitions as $definition) {
+      // We are only interested in the entity types that have the form class
+      // to unpublish content set on it.
+      if (!$definition->getFormClass('unpublish')) {
+        continue;
+      }
+
       $canonical_route = $collection->get('entity.' . $definition->id() . '.canonical');
       $route = new Route(
         // Path.
         $canonical_route->getPath() . '/unpublish',
         // Defaults.
         [
-          '_form' => ContentEntityUnpublishForm::class,
+          '_entity_form' => "{$definition->id()}.unpublish",
           '_title' => 'Unpublish',
         ],
         // Requirements.
         [
-          '_custom_access' => ContentEntityUnpublishForm::class . ':access',
+          '_custom_access' => ContentEntityUnpublishForm::class . '::access',
```

Comment:
Would it make sense here to create a new access handler? That way we could pass $definition->id() as a parameter to the access function and we wouldn't need to generate it with
`$entity_form = $routeMatch->getRouteObject()->getDefault('_entity_form');
    list ($entity_type_id, $operation) = explode('.', $entity_form);`

Reply:
No need for that reason. This is also what core does for the `_entity_form` handler for example. See `HtmlEntityFormController::getFormObject()`